pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,581 @@
+				Storage: &cdiv1.StorageSpec{},
+				Source: &cdiv1.DataVolumeSource{
+					PVC: &cdiv1.DataVolumeSourcePVC{
+						Namespace: namespace,
+						Name:      name,
+					},
+				},
+			},
+		}
+		if vol.Size != nil {
+			dvt.Spec.Storage.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceStorage: *vol.Size,
+			}
+		}
+		vm.Spec.DataVolumeTemplates = append(vm.Spec.DataVolumeTemplates, dvt)
+
+		vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+			Name: vol.Name,
+			VolumeSource: v1.VolumeSource{
+				DataVolume: &v1.DataVolumeSource{
+					Name: vol.Name,
+				},
+			},
+		})
+	}
+
+	return nil
+}
+
+func withPvcVolume(c *createVM, vm *v1.VirtualMachine) error {
+	for _, pvcVol := range c.pvcVolumes {
+		vol := pvcVolume{}
+		err := mapParams(PvcVolumeFlag, pvcVol, &vol)
+		if err != nil {
+			return err
+		}
+
+		if vol.Source == "" {
+			return flagErr(PvcVolumeFlag, "src must be specified")
+		}
+
+		namespace, name, err := splitPrefixedName(vol.Source)
+		if err != nil {
+			return flagErr(PvcVolumeFlag, "src invalid: %w", err)
+		}
+		if namespace != "" {
+			return flagErr(PvcVolumeFlag, "not allowed to specify namespace of pvc '%s'", name)
+		}
+
+		if vol.Name == "" {
+			vol.Name = name
+		}
+
+		vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+			Name: vol.Name,
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+						ClaimName: name,
+					},
+				},
+			},
+		})
+	}
+
+	return nil
+}
+
+func withBlankVolume(c *createVM, vm *v1.VirtualMachine) error {
+	for i, blankVol := range c.blankVolumes {
+		vol := blankVolume{}
+		err := mapParams(BlankVolumeFlag, blankVol, &vol)
+		if err != nil {
+			return err
+		}
+
+		if vol.Size == nil {
+			return flagErr(BlankVolumeFlag, "size must be specified")
+		}
+
+		if vol.Name == "" {
+			vol.Name = fmt.Sprintf("%s-blank-%d", vm.Name, i)
+		}
+
+		vm.Spec.DataVolumeTemplates = append(vm.Spec.DataVolumeTemplates, v1.DataVolumeTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vol.Name,
+			},
+			Spec: cdiv1.DataVolumeSpec{
+				Storage: &cdiv1.StorageSpec{
+					Resources: k8sv1.ResourceRequirements{
+						Requests: k8sv1.ResourceList{
+							k8sv1.ResourceStorage: *vol.Size,
+						},
+					},
+				},
+				Source: &cdiv1.DataVolumeSource{
+					Blank: &cdiv1.DataVolumeBlankImage{},
+				},
+			},
+		})
+
+		vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+			Name: vol.Name,
+			VolumeSource: v1.VolumeSource{
+				DataVolume: &v1.DataVolumeSource{
+					Name: vol.Name,
+				},
+			},
+		})
+	}
+
+	return nil
+}
+
+func withCloudInitData(c *createVM, vm *v1.VirtualMachine) error {
+	// Skip if cloudInitDisk is already present
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.Name == cloudInitDisk {
+			return nil
+		}
+	}
```

Comment:
I'd rather raise an error here to highlight to the caller that they can't add multiple cloud-init disks?